test/gtest/core/test_reduce.cc

```diff
@@ -235,3 +238,45 @@ TYPED_TEST(test_reduce, multiple_cuda_inplace) {
     TEST_DECLARE_MULTIPLE(UCC_MEMORY_TYPE_CUDA, TEST_INPLACE);
 }
 #endif
+
+template <typename T> class test_reduce_avg_order : public test_reduce<T> {
```

Comment:
lets also extend test_allreduce (not only reduce). Because for "reduce" we have 1 alg in tl/ucp while for allreduce we also have SRA which is modified by the PR as well.

Reply:
test_allreduce was also extended